### PR TITLE
Feature#18 conect tables db registers

### PR DESCRIPTION
**Pull request by xManuel007**
# Description

What did you implemented/Why did you implemented this:

- We modified the tables and connect with foreign keys.
- we added the company field so now we can connect company table with employees
- Also we added the field company in the CRUD of employees

# Screenshots

image

### CODE REVIEW

File: app/controllers/employees_controller.rb [hunk 1]
```diff
@@ -9,19 +9,23 @@ def show
...
   def new
     @employee = Employee.new
+    @companies = Company.select(:name, :id)
   end
 
   def create
-    @employee = Employee.new(employee_params)
-    
+    @company = Company.find_by(company_params)
+    puts @company.id
```
Comment (juperameza) [inline app/controllers/employees_controller.rb:17]: Is this a debugging put?
If it is you should delete it

File: app/controllers/employees_controller.rb [hunk 2]
```diff
@@ -9,19 +9,23 @@ def show
...
+    @employee = @company.employees.new(employee_params)
+    puts "sussesfull"
```
Comment (juperameza) [inline app/controllers/employees_controller.rb:19]: Same here

File: db/migrate/20230128041011_create_checks.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 class CreateChecks < ActiveRecord::Migration[7.0]
   def change
     create_table :checks do |t|
-      t.belongs_to :employee, index: true, foreign_key: true
+      t.belongs_to :employee, index: true, null: false, foreign_key: true
```
Comment (juperameza) [inline db/migrate/20230128041011_create_checks.rb:4]: You shouldn't modify an existing migration instead create a new one

**Review by juperameza — CHANGES_REQUESTED** [review]

**Review by AangelF — APPROVED** [review]

**Review by dagp-ucol — CHANGES_REQUESTED** [review]
As @juperameza said, you modified a lot of migrations, that is really bad practice, because you are deleting the code history of the database and also you will have a lot of problems in the future with the schema.

You can delete migrations with commands, but never manually or edit them.

File: db/migrate/20230128041011_create_checks.rb (hunk 3 above)
Comment (xManuel007) [reply db/migrate/20230128041011_create_checks.rb:4]: We create a new migration but with the same name, so maybe GitHub recognize like a modified file

File: app/controllers/employees_controller.rb (hunk 2 above)
Comment (xManuel007) [reply app/controllers/employees_controller.rb]: done

**Review by Reiyirak — APPROVED** [review]

**Review by juperameza — APPROVED** [review]
Lgtm just try to use different names for migrations

**Comment by oscaregr** [comment]
@dagp-ucol I destroyed the migrations and recreated them with the same name, maybe that's why git marks them as edited

**Review by dagp-ucol — APPROVED** [review]
I have seen what happened, other thing that can help is to specify in the commit description what changes you made. Can be short, but we can see what happened to certain files and understand more about the changes you made.

LGTM 🙂

**Pull request merged**